packages/tocco-util/src/actions/modules/actionHandlers/report.js

```diff
@@ -73,17 +74,23 @@ export function* handleReportGenerations(settingsModalId, generationsResponse) {
   const completed = yield call(sagaHelpers.checkStatusLoop, pollingUrl, 'in_progress')
   yield put(notifier.removeBlockingInfo(blockingInfoId))
   if (completed.body.status === 'completed') {
-    yield call(handleSuccessfulReport, completed)
+    yield call(handleSuccessfulReport, completed, submitAction)
   } else {
     yield call(handleFailedReport)
   }
 }
 
-export function* handleSuccessfulReport(completed) {
+export function* handleSuccessfulReport(completed, submitAction) {
   const outputJobId = completed.body.outputJobId
   const outputJob = yield call(fetchEntity, 'Output_job', outputJobId, {paths: ['document']})
   const {fileName, binaryLink} = outputJob.paths.document.value.value
-  yield call(download.downloadUrl, binaryLink, fileName)
+
+  if (submitAction === submitActions.DOWNLOAD) {
+    yield call(download.downloadUrl, binaryLink, fileName)
+  } else if (submitAction === submitActions.DISPLAY) {
+    yield call(download.openUrl, binaryLink)
+  }
+
```

Comment:
Same as above. Could `download.downloadUrl`, `download.openUrl` and both actions have their own constants like you used in the tests?